examples/connectivity/plot_extract_regions_canica_maps.py

```diff
@@ -0,0 +1,85 @@
+"""
+Regions extraction using Canonical ICA maps and functional connectomes
+======================================================================
+
+This example shows how to use :class:`nilearn.regions.RegionExtractor`
+to extract connected brain regions from whole brain ICA maps and
+use them to estimate a connectome.
+
+We used 20 resting state ADHD functional datasets from :func:`nilearn.datasets.fetch_adhd`
+and :class:`nilearn.decomposition.CanICA` for whole brain ICA maps.
+
+Please see the related documentation of :class:`nilearn.regions.RegionExtractor`
+for more details.
+"""
+import numpy as np
+from nilearn import datasets
+print(" -- Fetching ADHD resting state functional datasets -- ")
+adhd_dataset = datasets.fetch_adhd(n_subjects=20)
+func_filenames = adhd_dataset.func
+confounds = adhd_dataset.confounds
+
+from nilearn.decomposition import CanICA
+print(" -- Canonical ICA decomposition of functional datasets -- ")
+# Initialize canica parameters
+n_components = 8
+canica = CanICA(n_components=n_components, smoothing_fwhm=6.,
+                memory="nilearn_cache", memory_level=2,
+                random_state=0)
+
+canica.fit(func_filenames)
+components_img = canica.masker_.inverse_transform(canica.components_)
+
+from nilearn.regions import region_extractor
+print(" -- Extracting regions from ICA maps and timeseries signals -- ")
+extractor = region_extractor.RegionExtractor(
+    components_img, threshold=0.5, thresholding_strategy='ratio_n_voxels',
+    standardize=True, min_size=100)
+# Regions extraction from ICA maps
+extractor.fit()
+regions_extracted = extractor.regions_
+n_regions = regions_extracted.shape[3]
+print(" ====== Regions extracted ====== ")
+print(" -- Number of regions extracted from %d ICA components are %d -- " % (
+    n_components, n_regions))
+
+# Subjects timeseries signals extraction
+subjects_timeseries = []
+for img, confound in zip(func_filenames, confounds):
+    signals = extractor.transform(img, confounds=confound)
+    subjects_timeseries.append(signals)
+
+# Computing correlation coefficients of the timeseries signals
+correlation = []
+for subject_ts in subjects_timeseries:
+    corr = np.corrcoef(subject_ts.T)
+    correlation.append(corr)
```

Comment:
This can be simplified with a list comprehension:

```python
correlations = [np.corrcoeff(subject_ts.T) for subject_ts in subjects_timeseries]
```

I would use a plural (i.e. correlations rather than correlation) since this is a list.